app/journey/Journey.scala

```diff
@@ -73,7 +73,8 @@ final case class Journey(
     maybeEligibilityStatus:    Option[EligibilityStatus]       = None,
     debitDate:                 Option[LocalDate]               = None,
     ddRef:                     Option[String]                  = None,
-    maybeSaUtr:                Option[String]                  = None
+    maybeSaUtr:                Option[String]                  = None,
+    createdTTL:                LocalDateTime                   = LocalDateTime.now(ZoneOffset.UTC) // mongo is in UTC
```

Comment:
this will break existing journeys cos old journeys won't have this field - has that been taken into account already?

Reply:
Think the plan with SSTTP is to shutter for 15 mins to kill the current users but have now added some error handling that will add createdTTL to mongodb if it's missing

Reply:
if that was the case then maybe the recovering code is not needed after all - if the plan is to let the sessions expire then it should never get called in theory. If we leave it in we'd be creating a bit of work for ourselves in the future to remove it cos it'd only be a temporary thing. What do you think?-

Reply:
how about we merge this one and see if it does solve the problem in Staging, then remove the code if we still think it is not needed

Reply:
Haven't tried changing the instance number of anything other than the FE as this was showing up as more of a loss of the Journey after a couple of minutes and displaying a 'tech error page'. 1st guess was that the journey key was being held in a session cookie, that's why I tried a single instance, with no difference. 
Looking more on my laptop, it looks like mongo was deleting the session. This was because the test data is all set up for the year 2020, and so it should be deleted, so am trying to split the time that the Journey thinks it is and the time that mongo thinks it is from an auto-delete perspective.  (The tests are also built around one year, with things like interest calc results being compared with the expected results making it a 'fun' task to update them.  )

Reply:
maybe we were looking at different things then - I'll send you what I sent Kostas a while ago with what I found which led me to conclude that 1 instance of the payments-stub-protected might help the eligible/ineligible thing:

On the test-only page, after all the links the current session is shown if there is one. I compared what the session was when I got ineligible first and then when I eventually got eligible in the same session and I confirmed that the SAUTR was the same when it flipped but the debits part of the session changed. The reason for ineligibility was OldDebtIsTooHigh. Tracing where the debit info came from, it looks like it comes from payments-stubs-protected . For the same SAUTR I did a number of curl-microservice calls to get the debit info and confirmed that I sometimes got a different response. So, looking at the code it looks like the debit info is obtained here:
https://github.com/hmrc/payments-stubs-protected/blob/main/app/paymentsstubs/eligibilitystub/EligibilityStubController.scala#L89

and that does a get on a map defined here:

https://github.com/hmrc/payments-stubs-protected/blob/main/app/paymentsstubs/eligibilitystub/EligibilityStubController.scala#L36

This won’t work if there are more than one instance of that service running since the value returned will depend on which instance got the request since each will hold its own local copy of the map. Looking at the app-configs, payments-stubs-protected in staging does have 2 instances:
https://github.com/hmrc/app-config-staging/blob/main/payments-stubs-protected.yaml

but in dev and qa it’s only 1 which might explain why you don’t see that problem in that environment


The quick fix here then would be to set the number of instance of payments-stub-protected to 1 in staging, and the more sustainable solution going forward would be to stop using a map in a local val and to use mongo instead so that multiple instances can reference the same data